modules/videoio/cmake/init.cmake

```diff
@@ -0,0 +1,35 @@
+macro(add_backend backend_id cond_var)
+  if(${cond_var})
+    include("${CMAKE_CURRENT_LIST_DIR}/detect_${backend_id}.cmake")
+  endif()
```

Comment:
We should not change existed policy about `WITH_` / `HAVE_` variables.

Reply:
I think the difference is not too big and the proposed policy is less error prone. For example, building two configurations in the same build directory:
* build OpenCV with `WITH_` disabled by default and `HAVE_` enabled via command-line - feature is turned ON
* build OpenCV with `WITH_` disabled by default and none additional variables set - feature is still turned ON, because `HAVE_` is stored in cache and overrides `WITH_` option.

In the method which I propose, `HAVE_` is internal variable and will not be stored in cache, thus `WITH_` will actually disable or enable corresponding feature. If you pass `HAVE_` via command line in one build and do not pass it in consecutive builds, the feature will still be disabled because target-creation code will not be executed.

Reply:
I mean this:
```
if(WITH_${cond_var} OR HAVE_${cond_var})
    include(...)
endif()
```

and in `detect_xxx.cmake` (already done):

```
if(NOT HAVE_xxx)
  ... some auto-detection which work in most cases ...
endif()

... next auto detection attempt ...

if(HAVE_xxx)
  ... register dependencies with support of command-line vars: ...
  ... INCLUDE_DIRS, LIBRARIES, etc ...
endif()
```